### PR TITLE
feat: expose wtexture provider

### PR DESCRIPTION
**Pull request by asterwyx**
Expose wtexture provider.

### CODE REVIEW

File: src/server/qtquick/wtextureprovider.h [hunk 1]
```diff
@@ -0,0 +1,17 @@
+// Copyright (C) 2024 Yixue Wang <wangyixue@deepin.org>.
+// SPDX-License-Identifier: Apache-2.0 OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+#pragma once
+
+#include <wglobal.h>
+#include <qwtexture.h>
```
Comment (zccrs) [inline src/server/qtquick/wtextureprovider.h:6]: Can forward declaration?

File: src/server/qtquick/wtextureprovider.h [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+#include <QSGTextureProvider>
+
+WAYLIB_SERVER_BEGIN_NAMESPACE
+class WTextureProvider : public QSGTextureProvider
+{
+    Q_OBJECT
+public:
+    virtual QW_NAMESPACE::QWTexture *qwTexture() const = 0;
```
Comment (zccrs) [inline src/server/qtquick/wtextureprovider.h:15]: Needs QWBuffer too?

**Review by zccrs — CHANGES_REQUESTED** [review]

File: src/server/qtquick/wtextureprovider.h (hunk 1 above)
Comment (asterwyx) [reply src/server/qtquick/wtextureprovider.h:6]: Done.

File: src/server/qtquick/wtextureprovider.h (hunk 2 above)
Comment (asterwyx) [reply src/server/qtquick/wtextureprovider.h:15]: For TextureProvider in "woutputviewport.cpp", it doesn't store a copy of the source buffer. I'm not sure whether the source buffer will change or not after the texture is constructed.

Comment (asterwyx) [reply src/server/qtquick/wtextureprovider.h:15]: If one wants a QWBuffer, he can directly draw the texture on a buffer.

Comment (asterwyx) [reply src/server/qtquick/wtextureprovider.h:15]: Or we could ensure that "texture is consistent with buffer" at a higher level.

Comment (asterwyx) [reply src/server/qtquick/wtextureprovider.h]: done

**Review by zccrs — APPROVED** [review]

**Pull request merged**